crates/edr_napi/src/solidity_tests/config.rs

```diff
@@ -0,0 +1,99 @@
+use std::{collections::HashMap, path::PathBuf};
+
+use alloy_primitives::{address, Address};
+use edr_eth::U256;
+use forge::{
+    inspectors::cheatcodes::CheatsConfigOptions,
+    opts::{Env as EvmEnv, EvmOpts},
+};
+use foundry_compilers::ProjectPathsConfig;
+use foundry_config::{
+    cache::StorageCachingConfig, fs_permissions::PathPermission, FsPermissions, FuzzConfig,
+    GasLimit, InvariantConfig, RpcEndpoint, RpcEndpoints,
+};
+
+/// Solidity tests configuration
+#[derive(Clone, Debug)]
+pub(super) struct SolidityTestsConfig {
+    /// Project paths configuration
+    pub project_paths_config: ProjectPathsConfig,
+    /// Cheats configuration options
+    pub cheats_config_options: CheatsConfigOptions,
+    /// EVM options
+    pub evm_opts: EvmOpts,
+    /// Configuration for fuzz testing
+    pub fuzz: FuzzConfig,
+    /// Configuration for invariant testing
+    pub invariant: InvariantConfig,
+}
+
+/// Default address for tx.origin for Foundry
+///
+/// `0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38`
+const DEFAULT_SENDER: Address = address!("1804c8AB1F12E6bbf3894d4083f33e07309d1f38");
+
+impl SolidityTestsConfig {
+    /// Create a new `SolidityTestsConfig` instance
+    pub fn new(gas_report: bool) -> Self {
+        // Matches Foundry config defaults
+        let gas_limit: GasLimit = i64::MAX.into();
+        let evm_opts = EvmOpts {
+            env: EvmEnv {
+                gas_limit: gas_limit.into(),
+                chain_id: None,
+                tx_origin: DEFAULT_SENDER,
+                block_number: 1,
+                block_timestamp: 1,
+                ..Default::default()
```

Comment:
Are lints enabled? IIRC, a lint is enabled that warms when not doing this:
```suggestion
                ..EvmEnv::default()
```

Reply:
Yeah I double checked. This case isn't covered by the [`default_trait_access`](https://rust-lang.github.io/rust-clippy/v0.0.212/index.html#default_trait_access) lint, because the type is clear from the struct literal.